sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -499,4 +570,52 @@ private void addDiagnosticsOnTracerEvent(CosmosDiagnostics cosmosDiagnostics, Co
         this.addEvent("ClientCfgs", attributes,
             OffsetDateTime.ofInstant(clientSideRequestStatistics.getRequestStartTimeUTC(), ZoneOffset.UTC), context);
     }
+
+    private static void subscribe(Tracer tracer, CoreSubscriber<? super Object> actual) {
+        Context context = actual.currentContext().getOrDefault(REACTOR_TRACING_CONTEXT_KEY, null);
+        if (context != null) {
+            AutoCloseable scope = tracer.makeSpanCurrent(context);
+            try {
+                actual.onSubscribe(Operators.scalarSubscription(actual, DUMMY_VALUE));
```

Comment:
What is the logic behind this dummy value in subscribe. ?

Reply:
it will be ignored- it's this `ignored` value in 
```java
return propagatingFlux
            .flatMap(ignored -> publisher);
```

all magic is done by `AutoCloseable scope = tracer.makeSpanCurrent(context)`